src/ucp/core/ucp_mm.c

```diff
@@ -343,7 +354,12 @@ static ucs_status_t ucp_mem_map_common(ucp_context_h context, void *address,
             goto err_free_memh;
         }
     } else {
-        memh->mem_type     = ucp_memory_type_detect(context, address, length);
+        if (memory_type != UCS_MEMORY_TYPE_LAST) {
```

Comment:
maybe add to memory_type.h:
enum {
...
UCS_MEMORY_TYPE_LAST,
UCS_MEMORY_TYPE_UNKNOWN = UCS_MEMORY_TYPE_LAST
}